catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java

```diff
@@ -54,6 +69,50 @@ public void showNotification(CloudMessage cloudMessage) {
 		}
 	}
 
+	public static boolean initialize(Context context) {
+		if (isCloudNotificationsEnabled(context)) {
+			subscribeToFirebaseServices(context);
+			return true;
+		}
+		unsubscribeFromFirebaseServices(context);
+		return false;
+	}
+
+	@VisibleForTesting
+	public static void setIsCloudNotificationsEnabled(boolean isEnabled) {
+		isCloudNotificationsEnabled = isEnabled;
+	}
+
+	private static void subscribeToFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().subscribeToTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().subscribeToTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().subscribeToTopic(getCountryCode(context));
+	}
+
+	private static void unsubscribeFromFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getCountryCode(context));
+	}
+
+	private static boolean isCloudNotificationsEnabled(Context context) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		return sharedPreferences.getBoolean(SettingsActivity.SETTINGS_CLOUD_NOTIFICATIONS, false) && isCloudNotificationsEnabled;
+	}
+
+	private static String getLanguageCode(Context context) {
+		String languageCode = context.getResources().getConfiguration().locale.getCountry();
```

Comment:
`getCountry` = `US`
`getLanguage` = `en`

Please do some research on how the language code is standardized. See [getLanguage](https://developer.android.com/reference/java/util/Locale.html#getLanguage())